e2e/e2etests/test_migrate_tss.go

```diff
@@ -0,0 +1,182 @@
+	utxos, err := r.GetTop20UTXOsForTssAddress()
+	require.NoError(r, err)
+
+	var btcBalance float64
+	for _, utxo := range utxos {
+		btcBalance += utxo.Amount
+	}
+
+	btcTSSBalanceOld := btcBalance
+	// Use fixed fee of 0.01 for migration
+	btcBalance = btcBalance - 0.01
+	btcChain := chains.BitcoinRegtest.ChainId
+
+	//migrate btc funds
+	// #nosec G701 e2eTest - always in range
+	migrationAmountBTC := sdkmath.NewUint(uint64(btcBalance * 1e8))
+	msgMigrateFunds := crosschaintypes.NewMsgMigrateTssFunds(
+		r.ZetaTxServer.GetAccountAddress(0),
+		btcChain,
+		migrationAmountBTC,
+	)
+	_, err = r.ZetaTxServer.BroadcastTx(utils.FungibleAdminName, msgMigrateFunds)
+	require.NoError(r, err)
+
+	// Fetch migrator cctx for btc migration
+	migrator, err := r.ObserverClient.TssFundsMigratorInfo(r.Ctx, &observertypes.QueryTssFundsMigratorInfoRequest{
+		ChainId: btcChain})
+	require.NoError(r, err)
+	cctxBTCMigration := migrator.TssFundsMigrator.MigrationCctxIndex
+
+	// ETH migration
+	// Fetch balance of ETH TSS address
+	tssBalance, err := r.EVMClient.BalanceAt(context.Background(), r.TSSAddress, nil)
+	require.NoError(r, err)
+	ethTSSBalanceOld := tssBalance
+
+	tssBalanceUint := sdkmath.NewUintFromString(tssBalance.String())
+	evmChainID, err := r.EVMClient.ChainID(r.Ctx)
+	require.NoError(r, err)
+
+	// Migrate TSS funds for the eth chain
+	msgMigrateFunds = crosschaintypes.NewMsgMigrateTssFunds(
+		r.ZetaTxServer.GetAccountAddress(0),
+		evmChainID.Int64(),
+		tssBalanceUint,
+	)
+	_, err = r.ZetaTxServer.BroadcastTx(utils.FungibleAdminName, msgMigrateFunds)
+	require.NoError(r, err)
+
+	// Fetch migrator cctx for eth migration
+	migrator, err = r.ObserverClient.TssFundsMigratorInfo(
+		r.Ctx,
+		&observertypes.QueryTssFundsMigratorInfoRequest{ChainId: evmChainID.Int64()},
+	)
+	require.NoError(r, err)
+	cctxETHMigration := migrator.TssFundsMigrator.MigrationCctxIndex
+
+	cctxBTC := utils.WaitCCTXMinedByIndex(r.Ctx, cctxBTCMigration, r.CctxClient, r.Logger, r.CctxTimeout)
+	require.Equal(r, crosschaintypes.CctxStatus_OutboundMined, cctxBTC.CctxStatus.Status)
+
+	cctxETH := utils.WaitCCTXMinedByIndex(r.Ctx, cctxETHMigration, r.CctxClient, r.Logger, r.CctxTimeout)
+	require.Equal(r, crosschaintypes.CctxStatus_OutboundMined, cctxETH.CctxStatus.Status)
+
+	// Check if new TSS is added to list
+	allTss, err := r.ObserverClient.TssHistory(r.Ctx, &observertypes.QueryTssHistoryRequest{})
+	require.NoError(r, err)
+
+	require.Len(r, allTss.TssList, 2)
+
+	// Update TSS to new address
+	sort.Slice(allTss.TssList, func(i, j int) bool {
+		return allTss.TssList[i].FinalizedZetaHeight < allTss.TssList[j].FinalizedZetaHeight
+	})
+	msgUpdateTss := crosschaintypes.NewMsgUpdateTssAddress(
+		r.ZetaTxServer.GetAccountAddress(0),
+		allTss.TssList[1].TssPubkey,
+	)
+	_, err = r.ZetaTxServer.BroadcastTx(utils.FungibleAdminName, msgUpdateTss)
+	require.NoError(r, err)
+
+	// Wait for atleast one block for the TSS to be updated
+	time.Sleep(8 * time.Second)
```

Comment:
Would be cool to have a reliable `r.WaitForNextBlock(chainID)` helper

Reply:
Added something similar here 
https://github.com/zeta-chain/node/blob/afc91751c9570d25e649587919ba471700a7fa30/e2e/e2etests/helpers.go#L22

Will update to use the same once I merge that into develop